### DISCUSSION TITLE
NSS softoken now supports environment variables for configDir etc like softhsm

### DISCUSSION THREAD

**Discussion by space88man** [openssl-projects/pkcs11-provider#350] [Ideas]
FYI for the developers here: when asking on OpenSC for the equivalent of `pkcs11-module-init-args` for pkcs11-tool, a comment pointed out that NSS softoken now supports environment variables so there is less of a need to abuse `C_Initialize()`.

```
NSS_LIB_PARAMS="configDir=xxxx" openssl ....
```

https://github.com/OpenSC/OpenSC/issues/3049#issuecomment-1966095710

https://github.com/nss-dev/nss/commit/c0bbdc4ed7b3520c44600f0e254887d2ae938325

Tested to be working with 3.96 on Fedora 39.

**Comment by Jakuje** [comment]
This was implemented in #436 -- we are still waiting for the updated NSS to land in centos9stream, but it simplified the setup quite a bit.

**Comment by simo5** [comment]
Closing as done
